src/likelihoods/negativebinomial.jl

```diff
@@ -0,0 +1,25 @@
+"""
+    NegativeBinomialLikelihood(l=logistic; successes::Real=1)
+
+Negative binomial likelihood with number of successes `successes`.
+
+```math
+    p(k|f) = \\frac{\\Gamma(k+successes)}{k! \\Gamma(successes)} l(f)^successes (1 - l(f))^k
+```
+On calling, this returns a negative binomial distribution with `successes` successes and probability of success equal to `l(f)`.
+Note the distinction of the role of parameter `successes` to the parameter `r` in the Wikipedia definition.
```

Comment:
Maybe you can write that to get the equivalent of the Wikipedia definition, one just needs to define p = 1 - p

Reply:
Maybe this is slightly awkward, but the reason this is different from the Wikipedia definition is because Distributions.jl chose to use a different convention. I wonder whether it is better to be consistent with Wikipedia or within the Julia ecosystem.

Reply:
It's a very annoying one. I don't know why Distributions.jl went with this parameterization, maybe it's inherited from R...
Anyway I think we should be consistent with Distributions.jl but we just need to have a warning in big red letters in the docs and how to convert to the Wikipedia parameterization.

Reply:
Shall I add a boldface **CAUTION** to the docstring? On a related note, any thoughts about the initial comment in this PR?

Reply:
> Maybe you can write that to get the equivalent of the Wikipedia definition, one just needs to define p = 1 - p

Yes, I could add that, but note that at least for the logistic link this corresponds to an unidentifiable sign change of the latent GP.

Reply:
> Shall I add a boldface CAUTION to the docstring?

You can add a `!!! warning` admonition, this will show up prominently in the docs and the REPL: https://juliadocs.github.io/Documenter.jl/stable/showcase/#Admonitions